contacts.py

```diff
@@ -370,8 +370,24 @@ def update(self, updating_token, updates, floating_seconds, serial_number):
         else:
             return False
 
-    def sorted_list_by_time_and_serial_number_range(self, since, until):
-        return self.sorted_list_by_time_and_serial_number[self.sorted_list_by_time_and_serial_number.bisect_left((since, 0)):self.sorted_list_by_time_and_serial_number.bisect_left((until, 0))]
+    #TODO-42
+    def _sorted_idx(self, since):
+        return self.sorted_list_by_time_and_serial_number.bisect_left((since, 0))
+
+    #TODO-42
+    def max_until(self, since, until, maximum_results):
+        left = self._sorted_idx(since)
+        if left + maximum_results >= len(self.sorted_list_by_time_and_serial_number):
+            return until
+        else:
+            return self.sorted_list_by_time_and_serial_number[left + maximum_results][0]
+
+    def sorted_list_by_time_and_serial_number_range(self, since, until, maximum_results):
+        # This is only used by /sync as it doesn't filter by prefix or bbox
+        since_idx = self._sorted_idx(since)
+        until_idx = self._sorted_idx(until)
+        right_idx = min(since_idx + maximum_results, until_idx) if maximum_results else until_idx
```

Comment:
?? doesn't line 323 pass None as maximum_results?  doesn't you want the default value to be 0?   Since line 389 will then try to add None to a numeric index.

Reply:
Shouldnt' "None" is False

Reply:
So 389 wlll take the "Else" case (right of the line)

Reply:
ah, missed the if